src/bootstrap/setup.rs

```diff
@@ -107,6 +108,19 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let include_path = profile.include_path(src_path);
     println!("`x.py` will now use the configuration at {}", include_path.display());
 
+    let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
+    let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
+    let toolchain_link =
+        Command::new("rustup").args(&["toolchain", "link", "stage1", &stage_path[..]]).output();
+    if toolchain_link.is_err() {
+        println!("WARNING: `rustup`failed to link stage 1 build to `stage1` toolchain");
+        println!(
+            "To manually link stage 1 build to `stage1` tool chain, run:\n
+            `rustup toolchain link stage1 build/{}/stage1`",
+            build.rustc_target_arg()
+        );
+    }
```

Comment:
It would be nice to let people know x.py did this, so they know to use it in the future.
```suggestion
    } else {
        println!("Added `stage1` rustup toolchain; use `cargo +stage1 build` to run a newly-built toolchain");
    }
```

Reply:
Perhaps a command that would be less disruptive if it was executed immediately on the repository they are in, which could cause amusing errors? Or perhaps additional context to when and where they should run it. There are actually some people who first try Rust by building from the source, so we shouldn't assume they are an old hand.

Reply:
> Perhaps a command that would be less disruptive if it was executed immediately on the repository they are in, which could cause amusing errors?

Ideally we would fix this with https://github.com/rust-lang/rust/issues/76444 ... but I agree in the meantime we should suggest something else. Maybe add "on a separate rust project" after the `cargo build` snippet?

Reply:
Could change `use` to `try`